includes/Modules/Analytics_4.php

```diff
@@ -307,6 +290,27 @@ function ( $audience ) {
 						$this->resource_data_availability_date->reset_resource_date( $unavailable_audience_name, Resource_Data_Availability_Date::RESOURCE_TYPE_AUDIENCE );
 					}
 				}
+			}
+		);
+
+		$this->get_settings()->on_change(
+			function ( $old_value, $new_value ) {
+				// Ensure that the data available state is reset when the property ID or measurement ID changes.
+				if ( $old_value['propertyID'] !== $new_value['propertyID'] || $old_value['measurementID'] !== $new_value['measurementID'] ) {
+					$this->reset_data_available();
+					$this->custom_dimensions_data_available->reset_data_available();
+
+					$available_audiences = $old_value['availableAudiences'] ?? array();
```

Comment:
`$available_audiences` will need to be retrieved from `$this->audience_settings` here, as it's no longer in the module settings.